app/views/partners/children/show.html.erb

```diff
@@ -59,7 +59,11 @@
                     <dd><%= @child.health_insurance %></dd>
 
                     <dt>Item needed:</dt>
-                    <dd><%= current_partner.organization.item_id_to_display_string_map[@child.item_needed_diaperid] %></dd>
+                    <ul>
+                      <% @child.items.each do |item| %>
```

Comment:
One thing to be mindful of is how big a collection should be. This will show all of the items. So if there is a chance that there will be thousands of items, it is useful to try to limit it. It might be fine in this case though.